lib/suggestion-list.coffee

```diff
@@ -0,0 +1,112 @@
+{Model} = require 'theorist'
+{Emitter, CompositeDisposable} = require 'event-kit'
+
+module.exports =
+class SuggestionList extends Model
+  constructor: ->
+    @emitter = new Emitter
+    @subscriptions = new CompositeDisposable
+    # Allow keyboard navigation of the suggestion list
+    @subscriptions.add(atom.commands.add 'autocomplete-suggestion-list',
+      'autocomplete-plus:confirm': @confirmSelection,
+      'autocomplete-plus:select-next': @selectNext,
+      'autocomplete-plus:select-previous': @selectPrevious,
+      'autocomplete-plus:cancel': @cancel
+    )
+
+  addKeyboardInteraction: ->
+    @removeKeyboardInteraction()
+    keys =
+      'escape': 'autocomplete-plus:cancel'
+
+    completionKey = atom.config.get('autocomplete-plus.confirmCompletion') || ''
+    navigationKey = atom.config.get('autocomplete-plus.navigateCompletions') || ''
+
+
+    keys['tab'] = 'autocomplete-plus:confirm' if completionKey.indexOf('tab') > -1
+    keys['enter'] = 'autocomplete-plus:confirm' if completionKey.indexOf('enter') > -1
+
+    if @items?.length > 1 and navigationKey == 'up,down'
+      keys['up'] =  'autocomplete-plus:select-previous'
+      keys['down'] = 'autocomplete-plus:select-next'
+    else
+      keys['ctrl-n'] = 'autocomplete-plus:select-next'
+      keys['ctrl-p'] = 'autocomplete-plus:select-previous'
+
+    @keymaps = atom.keymaps.add(
```

Comment:
Seems weird to build the keymap programmatically. Why not just use a CSON file on disk for the package? Perhaps our primitive UI for editing keymaps forces you to go through the config API for the confirm key, requiring this approach. But it's unfortunate.